tools/src/main/scala/scala/scalanative/codegen/ResourceEmbedder.scala

```diff
@@ -28,11 +29,38 @@ private[scalanative] object ResourceEmbedder {
       classpathDirectory: VirtualDirectory
   )
 
+  private final val ScalaNativeExcludePattern: String = "/scala-native/**"
+
   def apply(config: Config): Seq[Defn.Var] = Scope { implicit scope =>
     val classpath = config.classPath
+    val fs = FileSystems.getDefault()
+    val includePatterns =
+      config.compilerConfig.resourceIncludePatterns.map(p =>
+        fs.getPathMatcher(s"glob:$p")
+      )
+    val excludePatterns =
+      (config.compilerConfig.resourceExcludePatterns :+ ScalaNativeExcludePattern)
+        .map(p => fs.getPathMatcher(s"glob:$p"))
 
     implicit val position: Position = Position.NoPosition
 
+    val notInIncludePatterns =
+      s"Not matched by any include path ${includePatterns.mkString}"
+    type IgnoreReason = String
+
+    /** If the return value is defined, the given path should be ignored. If
+     *  it's None, the path should be included.
+     */
+    def shouldIgnore(path: Path): Option[IgnoreReason] =
+      includePatterns.find(_.matches(path)).fold(Option(notInIncludePatterns)) {
+        includePattern =>
+          excludePatterns
+            .find(_.matches(path))
+            .map(excludePattern =>
+              s"Matched by $includePattern by excluded by $excludePattern"
```

Comment:
Typo here? Also it might be worth to put patterns in quotes, so it's easier to spot empty pattern or containing whitespaces.  
```suggestion
              s"Matched by '$includePattern', but excluded by '$excludePattern'"
```

Reply:
Also I don't think we have a test case which would trigger this path

Reply:
https://github.com/tanishiking/scala-native/blob/0f99a2a2861e5aa647595469d6ea0e80181bae73/scripted-tests/run/resource-embedding/build.sbt#L78-L79

Actually, in project G, include pattern matches with `exclude/b.txt` but it will be excluded because of `exclude/b.txt`.
I updated the comments for that tests https://github.com/scala-native/scala-native/pull/3562/commits/1b9c8defaedf298b24374b07115313a665c7e83b